manager/orchestrator/restart/restart.go

```diff
@@ -214,8 +219,8 @@ func (r *Supervisor) shouldRestart(ctx context.Context, t *api.Task, service *ap
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	restartInfo := r.history[instanceTuple]
-	if restartInfo == nil {
+	restartInfo := r.historyByService[t.ServiceID][instanceTuple]
```

Comment:
I believe the TODO is saying that `service` should not be passed in as a function parameter. I don't see how this function could work without being aware of the service ID.